types/tuple.hh

```diff
@@ -123,7 +141,18 @@ public:
     const std::vector<data_type>& all_types() const {
         return _types;
     }
-    std::vector<bytes_view_opt> split(bytes_view v) const;
+    std::vector<bytes_opt> split(FragmentedView auto v) const {
```

Comment:
This is a pessimization since we will now construct a bytes_opt for every element, possibly allocating storage. I think it's fine for now, since tuples are rare. We should revisit in the future and return a vector<fragmented view>.

Reply:
This looks like a regression at a glance, but `split` was actually only used for two things:
1. Converting the returned `vector<bytes_view_opt>` to `vector<bytes_opt>`.
2. Getting the n-th element; I replaced this use with a separate function (`get_nth_tuple_element`).

Reply:
Ok.